keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -243,147 +242,161 @@ A `Bucket` is not deleted if it is bound to a `BucketRequest`.
+1. `credentials`: (omitted if `serviceAccount` is used) a `SecretReference` containing the name, namespace and UID of a secret containing credentials that will be used by the driver to provision the bucket.
 1. `policyActionsConfigMapData`: encoded data, known to the driver, and defined by the admin when creating a `BucketAccessClass`. Contains a set of provisioner/platform defined policy actions to a given user identity. Contents of the ConfigMap that the *policyActionsConfigMap* field in the `BucketAccessClass` refers to. A sample value of this field could look like:
 ```
    {“Effect”:“Allow”,“Action”:“s3:PutObject”,“Resource”:“arn:aws:s3:::profilepics/*“},
    {“Effect”:“Allow”,“Action”:“s3:GetObject”,“Resource”:“arn:aws:s3:::profilepics/*“},
    {“Effect”:“Deny”,“Action”:“s3:*“,”NotResource”:“arn:aws:s3:::profilepics/*“}
 ```
-1. `principal`: username/access-key for the object storage provider to uniquely identify the user who has access to this bucket. This value is minted by the provisioner (and set in the BA) when the `BucketAccessRequest` omits a ServiceAccount. There are several use cases involving `principal`: 
-   - Not relevant if serviceAccount is supplied.
-   - For brownfield cases when there is already a storage-side principal that you want to use for access, the admin can supply this, the driver will (possibly) enable access for that principal to the bucket in question and will return credentials.
-   - For greenfield cases, this could be populated by the driver. Maybe so that the driver can request deletion of this principal during deprovisioning
-1. `parameters`:  A map of string:string key values.  Allows admins to control user and access provisioning by setting provisioner key-values. Copied from `BucketAccessClass`. 
-1. `accessGranted`: if true access has been granted to the bucket. If false then access to the bucket has not been granted.
+10. `parameters`:  A map of string:string key values.  Allows admins to control user and access provisioning by setting provisioner key-values. Copied from `BucketAccessClass`. 
+11. `accessGranted`: if true access has been granted to the bucket. If false then access to the bucket has not been granted.
+12. `credentials`: a `SecretReference` to the sidecar-generated Secret containing access credentials. This secret exists in the provisioner’s namespace, is read by the cosi-node-adapter, and written to the secret mount defined in the app pod's `csi-driver spec`.
```

Comment:
We need a mechanism to request a bucket while provisioning a pod, just like claims are requested when provisioning pods. Since it is impossible to know the secret name upfront, while provisioning the pod, we need a CSI driver that will do it

Reply:
dup of above thread, closing this one